resource/src/impls/ash.rs

```
@@ -0,0 +1,545 @@
+        Format::ASTC_10x5_SRGB_BLOCK => Astc10x5SrgbBlock,
+        Format::ASTC_10x6_UNORM_BLOCK => Astc10x6UnormBlock,
+        Format::ASTC_10x6_SRGB_BLOCK => Astc10x6SrgbBlock,
+        Format::ASTC_10x8_UNORM_BLOCK => Astc10x8UnormBlock,
+        Format::ASTC_10x8_SRGB_BLOCK => Astc10x8SrgbBlock,
+        Format::ASTC_10x10_UNORM_BLOCK => Astc10x10UnormBlock,
+        Format::ASTC_10x10_SRGB_BLOCK => Astc10x10SrgbBlock,
+        Format::ASTC_12x10_UNORM_BLOCK => Astc12x10UnormBlock,
+        Format::ASTC_12x10_SRGB_BLOCK => Astc12x10SrgbBlock,
+        Format::ASTC_12x12_UNORM_BLOCK => Astc12x12UnormBlock,
+        Format::ASTC_12x12_SRGB_BLOCK => Astc12x12SrgbBlock,
+        _ => panic!("Format {:?} isn't supported by the hal backend", f),
+    }
+}
+
+impl<V> Device for ash::Device<V>
+where
+    V: FunctionPointers,
+    ash::Device<V>: DeviceV1_0,
+{
+    type Sampler = vk::Sampler;
+    type Buffer = vk::Buffer;
+    type UnboundBuffer = vk::Buffer;
+    type BufferView = vk::BufferView;
+    type Image = vk::Image;
+    type UnboundImage = vk::Image;
+    type ImageView = vk::ImageView;
+
+    fn create_buffer(
+        &self,
+        info: buffer::CreateInfo,
+    ) -> Result<Self::UnboundBuffer, memory::OutOfMemoryError> {
+        let info = vk::BufferCreateInfo {
+            s_type: vk::StructureType::BufferCreateInfo,
+            p_next: ptr::null(),
+            flags: vk::BufferCreateFlags::empty(),
+            size: info.size,
+            usage: vk::BufferUsageFlags::from_flags_truncate(info.usage.bits()),
+            sharing_mode: match info.sharing {
+                SharingMode::Exclusive => vk::SharingMode::Exclusive,
+            },
+            queue_family_index_count: 0,
+            p_queue_family_indices: ptr::null(),
+        };
+
+        unsafe {
+            <Self as DeviceV1_0>::create_buffer(self, &info, None)
+        }.map_err(|_| memory::OutOfMemoryError::OutOfDeviceMemory)
```

You should match concrete error variants and panic on unexpected error variants.